### PR TITLE
Support exception chaining

### LINKED ISSUE #190: Support chained exceptions
In Ruby 2.1.0, [`Exception#cause`](http://www.ruby-doc.org/core-2.1.0/Exception.html#method-i-cause) was added. While @dcramer informed me in #sentry on irc.freenode.net that the Sentry server already supports chained exceptions, the Ruby client does not. I propose that we add support for chained exceptions to the Ruby client. We could just detect whether an exception responds to `#cause` and, if so, add the cause's message and backtrace to the exception report.

I'm just opening this for discussion now. I may propose this work as a "10% time" project at my current client or try to work on it myself as early as this weekend.

### PR DESCRIPTION
**Pull request by javawizard**
Fixes #190

### CODE REVIEW

**Comment by dcramer** [comment]
Unrelated to the code, but any idea which version of Ruby started offering chaining? I didn't actually realize it was even a feature.

**Comment by dcramer** [comment]
Ignore that comment, I'm an idiot. The ticket actually points it out :)

**Comment by nateberkopec** [comment]
Thanks! Will take a look.

**Comment by javawizard** [comment]
Awesome. Just pushed up a fix to not check reporting when causes aren't supported.

File: lib/raven/event.rb [hunk 1]
```diff
@@ -159,14 +159,25 @@ def self.from_message(message, options = {})
...
+      evt.interface(:exception) do |exc_int|
+        exceptions = [exc]
+        while exc.respond_to?(:cause) && exc.cause
+          exceptions << exc.cause
+          exc = exc.cause
+        end
+        exceptions.reverse!
+
+        exc_int.values = exceptions.map do |exc|
+          SingleExceptionInterface.new do |int|
```
Comment (nateberkopec) [inline lib/raven/event.rb]: Hmmm, I'm not really a fan of this name/a new class. Why can't you just use multiple instances of ExceptionInterface?

File: lib/raven/interfaces/exception.rb [hunk 2]
```diff
@@ -4,15 +4,12 @@ module Raven
   class ExceptionInterface < Interface
 
     name 'exception'
-    attr_accessor :type
-    attr_accessor :value
-    attr_accessor :module
-    attr_accessor :stacktrace
+    attr_accessor :values
```
Comment (nateberkopec) [inline lib/raven/interfaces/exception.rb]: Now I can't find the old documentation for how Exception interfaces are supposed to work with Sentry. I thought that info was here: http://sentry.readthedocs.org/en/latest/developer/interfaces/ but it isn't anymore. @dcramer where'd that go?

Comment (dcramer) [inline lib/raven/interfaces/exception.rb]: Docs should be fixed, but this looks correct to me (values is a list of exception instances)

**Pull request merged**